optuna/visualization/_regret_bound.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+from typing import NamedTuple
+
+import tqdm
+
+import optuna
+from optuna._experimental import experimental_func
+from optuna.logging import get_logger
+from optuna.study.study import Study
+from optuna.terminator.erroreval import BaseErrorEvaluator
+from optuna.terminator.erroreval import CrossValidationErrorEvaluator
+from optuna.terminator.improvement.evaluator import BaseImprovementEvaluator
+from optuna.terminator.improvement.evaluator import DEFAULT_MIN_N_TRIALS
+from optuna.terminator.improvement.evaluator import RegretBoundEvaluator
+from optuna.visualization._plotly_imports import _imports
+
+
+if _imports.is_successful():
+    from optuna.visualization._plotly_imports import go
+
+_logger = get_logger(__name__)
+
+
+PADDING_RATIO = 0.05
+OPACITY = 0.25
+
+
+class _RegretBoundInfo(NamedTuple):
+    trial_numbers: list[int]
+    regret_bounds: list[float]
+    errors: list[float] | None
+
+
+@experimental_func("3.2.0")
+def plot_regret_bound(
```

Comment:
Shall we rename this function to `plot_improvement` or `plot_terminator_improvement`?

Indeed, the argument is not necessarily a `RegretBoundEvaluator` but an `ImprovementEvaluator`.

If we rename, almost all occurrences of "regret bound" in this file should be renamed as well.

Reply:
Thank you for your suggestion. I changed the function name to `plot_terminator_improvement`, and also updated all occurrences of "regret bound" to "improvement". Could you take another look?